enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,205 @@
+   * New Windows VM will be created.
+   * The upgraded WMCO instance will configure Windows VM with containerd as the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+2) BYOH ( Bring Your Own Host)
+   * upgrade process will uninstall kubelet, kube-proxy, CNI, and hybrid-overlay components that were installed by WMCO.
+   * Windows OS-specific configurations like HNS-Network that were created by WMCO instance would be deleted.
+   * Once cleanup is done, WMCO instance will install containerd along with kubelet, kube-proxy, CNI and
+     hybrid-overlay.
+   * Kuebelet will start as a service. Containerd will be the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because Docker runtime support won't be available in kubelet.
+  As we don't support downgrade, reverting to an older version is not possible. To overcome
+  any issues, we introduce this feature as part of the 5.0 community operator so that this feature
+  can be well tested before we make it default in WMCO 6.0 release. We will use two test jobs to test all e2e test cases
+  in the 4.10 community branch(OKD) and 4.10 branch(ccm OCP)to capture any regression.
+  This will help us to identify any issues well before containerd becomes default runtime.
+* If we run into any issue in containerd, we work closely with the containerd open source community and try
+  to resolve it as soon as possible.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this
+  gets merged, if Windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. To overcome this issue, the user should log in to Windows VM and install `ctr or crictl`
+  command-line tool. Users can use one of these tools to pull the container image manually. Upon successful image download,
+  the user can create pods.
+* Currently Windows_exporter has been used to collect metrics from Windows node. we do
+  have containerd support in https://github.com/prometheus-community/Windows_exporter/releases/tag/v0.16.0
+  As of now, there is no feature parity mismatch between Docker and containerd.
+
+### Test Plan
+
+We will use two test jobs OKD and existing ccm OCP to test the WMCO community operator. We will pass the WMCO CLI option
+in the test jobs so that containerd will be enabled as the runtime. All exiting e2e tests can be run on
+one or two platforms. This will cover all the existing test cases. We should make sure there is no regression due
+to containerd runtime. Containerd is agnostic to the platform so testing in any platform should be fine.
```

Comment:
This is fine for the community branch what what about master?

Reply:
For the main branch, we use the ccm OCP test job to test it. I hope that is enough to test it until we make it default in 4.11. 
What do you think?

Reply:
That works for me, this should be cloud agnostic, so as long as we have at least one job testing this in master I am good with that.

Reply:
Do we need any vSphere coverage given that is the main platform for Windows use cases?